ocaml/xapi/xapi_pool_helpers.ml

```diff
@@ -112,6 +116,9 @@ let update_allowed_operations ~__context ~self : unit =
 (** Add to the Pool's current operations, call a function and then remove from the
     current operations. Ensure the allowed_operations are kept up to date. *)
 let with_pool_operation ~__context ~self ~doc ~op f =
+  if Db.Pool.get_is_psr_pending ~__context ~self:(Helpers.get_pool ~__context)
```

Comment:
Wouldn't the assert_operation_valid below make this one redundant too?

Reply:
Or is it becasue PSR is more like HA: instead of one long running API call you have something that starts it

Reply:
PSR isn't a 'proper' `pool_operation` so isn't in the `valid_operations` table, therefore we have to add the check specifically. However I'll move the check to `assert_operation_valid` because that makes more sense (this won't affect  behaviour anywhere else because this is the only usage)